src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(19);
+        user.setLogin("defaultLogin");
+        user.setPassword("defaultPassword");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        user.setLogin("");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setLogin("abc");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setLogin("abcdf");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        String testLogin = "login123";
+        user.setLogin(testLogin);
+        registrationService.register(user);
+        assertTrue(Storage.people.contains(user));
+        setUp();
+        user.setLogin(testLogin);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_validLogin_Ok() {
+        user.setLogin("someValidLogin");
+        registrationService.register(user);
```

Comment:
corner case 6 has to be tested
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/4b5b0667-efc7-46df-99b6-0dcc51664be1)